apps/sparo-lib/src/services/GitService.ts

```diff
@@ -438,6 +438,66 @@ Please specify a directory on the command line
     return objectType;
   }
 
+  public getCurrentBranch(): string {
+    const currentBranch: string = this.executeGitCommandAndCaptureOutput({
+      args: ['branch', '--show-current']
+    }).trim();
+    return currentBranch;
+  }
+
+  /**
+   * Check existence for a list of branch name
+   */
+  public checkRemoteBranchesExistenceAsync = async (
+    remote: string,
+    branches: string[]
+  ): Promise<Record<string, boolean>> => {
+    this._terminalService.terminal.writeDebugLine(`Checking branches: ${branches.join(',')}`);
+    const ret: Record<string, boolean> = {};
+    await Async.forEachAsync(branches, async (branch: string) => {
+      const isExists: boolean = await this.checkRemoteBranchExistenceAsync(remote, branch);
+      ret[branch] = isExists;
+    });
+    return ret;
+  };
+
+  /**
+   * Check existence for one branch name.
+   *
+   * Function "checkRemoteBranchesExistenceAsync" is preferred if you are going to check a list of branch name.
+   */
+  public checkRemoteBranchExistenceAsync = async (remote: string, branch: string): Promise<boolean> => {
+    const gitPath: string = this.getGitPathOrThrow();
+    const currentWorkingDirectory: string = this.getRepoInfo().root;
+    const childProcess: child_process.ChildProcess = Executable.spawn(
+      gitPath,
+      ['ls-remote', '--exit-code', remote, branch],
+      {
+        currentWorkingDirectory,
+        stdio: ['ignore', 'pipe', 'pipe']
+      }
+    );
+    if (!childProcess.stdout || !childProcess.stderr) {
+      this._terminalService.terminal.writeDebugLine(`Failed to spawn git process, fallback to spawnSync`);
```

Comment:
This seems like a strange condition. Under what circumstances would `spawn()` fail but `spawnSync()` would not?

Reply:
I feel like if `childProcess.stdout` does not exist for some reason, something is super wrong and the operation should fail, rather than retrying with a different code path that rarely ever gets any coverage.

Reply:
Also, general wisdom is that `async` spawn is more reliable than the synchronous version, just slightly more complicated to implement and debug.